R/centrality.R

```diff
@@ -499,27 +500,38 @@ arpack_defaults <- list(
 #' }
 #' @family arpack
 #' @export
-arpack <- function(func, extra = NULL, sym = FALSE, options = arpack_defaults,
+arpack <- function(func, extra = NULL, sym = FALSE, options = arpack_defaults(),
                    env = parent.frame(), complex = !sym) {
+
+  eval_try <- rlang::eval_tidy(options)
```

Comment:
What do we need `eval_tidy` for? Can't we just force, and if the user has passed a function, we call it?

Reply:
We only need to catch the case where the argument is `arpack_defaults` and a function. I'm using `rlang::call_args(rlang::current_call())[["options"]]` on the following line hence the rlang here, especially as rlang is a dependency.

Now, maybe this whole "hack" is a bad idea.